submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,106 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal{
+   constructor(species,name,gender,legs,friends,saying){
+      this.species = species,
+      this.name = name,
+      this.gender = gender,
+      this.legs = legs,
+      this.friends = friends,
+      this.saying = saying
+   }
+
+   getProperty(property){
+      if(this[property]){
+         return `<b>${property}:</b> ${this[property]}`;
+      }else{
+         return `<b>${property}:</b> ${this.species} doesn't have`;
+      }
+   }
+};
+
+class Dog extends Mammal{
+   constructor(name,gender,friends,saying){
+      super('dog',name,gender,4,friends,saying || 'guf');
+   }
+};
+
+class Cat extends Mammal{
+   constructor(name,gender,friends,saying){
+      super('cat',name,gender,4,friends,saying || 'meow');
+   }
+
+   getSaying(){
+      return this.saying || 'meow';
+   }
+}
+
+class Human extends Mammal{
+   constructor(name,gender,friends,saying){
+      super('human',name,gender,2,friends,saying);
+      this.hands = 2;
+      this.saying = saying || `Hello! My name is ${this.name}`;
+   }
+}
+
+class Woman extends Human{
+   constructor(name,friends,saying){
+      super(name,'female',friends,saying);
+   }
+}
+
+class Man extends Human{
+   constructor(name,friends,saying){
+      super(name,'male',friends,saying);
+   }
+}
+
+class CatWoman extends Woman{
+   constructor(name,friends){
+      super(name,friends,Cat.prototype.getSaying());
+   }
+};
+
+const cat = new Cat('Murzic','male',['Elza','Nicky']);
+const dog = new Dog('Lucy','female',['Alla','Dima'],'guffyy');
+const man = new Man('Vlad',['Sergay','Anton'],'Hello everyone!');
+const woman = new Woman('Anna',['Sergay','Artem']);
+const catWoman = new CatWoman('Lily',['Batman']);
+
+const inhabitants = [man,woman,cat,dog,catWoman];
+
+const properties = ['name','species','gender','legs','hands','saying','friends'];
+let output = inhabitants.map(inhabitant=>
+   properties.map(property=> inhabitant.getProperty(property))
+   .join('\t')
+).join('\n');
```

Comment:
We need to separate concerns.

Please make self-presentation a method of a class so that in this exact place we do not need to know anything about classes' internals (this what OOP is importantly about) but instead we can do something like `dog.tellAbout()` or `inhabitants.map(inhabitant => inhabitant.tellAbout())......`

If done properly you will not have any output on hands for cats or dogs instead of `hands: cat doesn't have any`. Imagine, you have birds in your world, or we want to know something else about humans. How overwhelmed with unnecessary information the output would then be? Proper compliance with OOP principles helps to avoid this.